streaming/src/main/scala/org/apache/gearpump/streaming/task/TaskActor.scala

```diff
@@ -124,92 +100,49 @@ class TaskActor(val taskContextData : TaskContextData, userConf : UserConfig, va
+    sendMsgWithTimeOutCallBack(appMaster, register, 10, registerTaskTimeOut())
     system.eventStream.subscribe(taskContextData.appMaster, classOf[MetricType])
-
-    val graph = dag.graph
-    LOG.info(s"TaskInit... taskId: $taskId")
-    val outDegree = dag.graph.outDegreeOf(taskId.processorId)
-
-    if (outDegree > 0) {
-
-      val edges = graph.outgoingEdgesOf(taskId.processorId)
-
-      LOG.info(s"task: ${taskId} out degree is $outDegree, edge length: ${edges.length}")
-
-      this.partitioner = edges.foldLeft(MergedPartitioner.empty) { (mergedPartitioner, nodeEdgeNode) =>
-        val (_, partitioner, processorId) = nodeEdgeNode
-        val taskParallism = dag.processors.get(processorId).get.parallelism
-        mergedPartitioner.add(partitioner, taskParallism)
-      }
-
-      LOG.info(s"task: ${taskId} partitioner: $partitioner")
-
-      outputTaskIds = edges.flatMap {nodeEdgeNode =>
-        val (_, _, processorId) = nodeEdgeNode
-        val taskParallism = dag.processors.get(processorId).get.parallelism
-
-        LOG.info(s"get output taskIds, processorId: $processorId, parallism: $taskParallism")
-
-        0.until(taskParallism).map { taskIndex =>
-          TaskId(processorId, taskIndex)
-        }
-      }.toArray
-
-    } else {
-      //outer degree == 0
-      this.partitioner = null
-      this.outputTaskIds = Array.empty[TaskId]
-    }
-
-    this.flowControl = new FlowControl(taskId, outputTaskIds.length, sessionId)
-    this.clockTracker = new ClockTracker(flowControl)
-
     context.become(waitForStartClock orElse stashMessages)
+
+    LOG.info(s"Task actor path: ${ActorUtil.getFullPath(system, self.path)}")
   }
 
   private def registerTaskTimeOut(): Unit = {
-    LOG.error(s"Task ${taskId} failed to register to AppMaster of application ${appId}")
+    LOG.error(s"Task ${taskId} failed to register to AppMaster of application $appId")
     throw new RestartException
   }
 
-  private def tryToSyncToClockService() : Unit = {
-    if (unackedClockSyncTimestamp == 0) {
-      appMaster ! UpdateClock(taskId, clockTracker.minClockAtCurrentTask)
-      needSyncToClockService = false
-      unackedClockSyncTimestamp = System.currentTimeMillis()
-    } else {
-      val current = System.currentTimeMillis()
-      if (current - unackedClockSyncTimestamp > CLOCK_SYNC_TIMEOUT_INTERVAL) {
-        appMaster ! UpdateClock(taskId, clockTracker.minClockAtCurrentTask)
-        needSyncToClockService = false
-        unackedClockSyncTimestamp  = System.currentTimeMillis()
-      } else {
-        needSyncToClockService = true
-      }
+  def minClockAtCurrentTask: TimeStamp = {
+    this.subscriptions.foldLeft(Long.MaxValue){ (clock, subscription) =>
+      Math.min(clock, subscription._2.minClock)
+    }
+  }
+
+  private def allowSendingMoreMessages(): Boolean = {
+    this.subscriptions.foldLeft(true) {
+      (status, subscrption) => status && subscrption._2.allowSendingMoreMessages()
```

Comment:
We can do this in future performance optimization when we find this is really an issue.